feat(user): use new progression components for recently played

wescopeland opened the pull request:
This PR:

* Migrates the user profile "Last N games played" to Blade.
* Expands the functionality of the game list item components on the Completion Progress page to be optionally collapsible and accept a slot.
* Uses the aforementioned list item components in the "Last N games played" section.

**Key Functional Differences**
* Only the first item with achievements is pre-expanded by default.
* Beaten award displays **are not feature-flagged.** We are so close to shipping that I didn't think it was worth the extra effort.

**Before**
![Screenshot 2023-11-05 at 12 56 22 PM](https://github.com/RetroAchievements/RAWeb/assets/3984985/fc087a84-a03a-4e51-8e04-d3b8820b1a96)

**After**
![Screenshot 2023-11-05 at 1 11 06 PM](https://github.com/RetroAchievements/RAWeb/assets/3984985/8729d799-e8bd-44a9-bce3-12c6af086219)


![Screenshot 2023-11-05 at 12 55 52 PM](https://github.com/RetroAchievements/RAWeb/assets/3984985/50d77313-e70c-4241-b575-862aa05bec89)

**Video**

https://github.com/RetroAchievements/RAWeb/assets/3984985/86403855-47a7-4c6c-ba44-13dcfbf18ffc

Jamiras commented:
> Only the first item with achievements is pre-expanded by default.

I find this behavior unexpected. If the first item doesn't have achievements, I wouldn't expect anything to be expanded. Once something becomes not-the-most-recent, it should be given less attention.

Jamiras commented:
I like the fade-in of the achievement when expanding, but I don't like the fact that they scroll up as they fade in. 
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/316f2c05-524a-46a5-9c99-58eb50b013d9)

If anything, they should scroll down to mimic the control expanding.

wescopeland commented:
> I find this behavior unexpected. If the first item doesn't have achievements, I wouldn't expect anything to be expanded. Once something becomes not-the-most-recent, it should be given less attention.

No problem. After stepping away from it for a little while and coming back, I can see how this is unexpected. I've adjusted this in latest so that only the first item will be pre-expanded if it has achievements.

> I like the fade-in of the achievement when expanding, but I don't like the fact that they scroll up as they fade in. If anything, they should scroll down to mimic the control expanding.

That makes sense. I've adjusted this in latest. The change is here: https://github.com/RetroAchievements/RAWeb/pull/1989/commits/1703827acc0383880abd72094b41dda7ce73abb6#diff-995fc2eec60ca841b104cfadf3ac52b6c74fb5ec5e9a132a884deabc37856e5dR111 and was done by swapping `translate-y-1.5` to `-translate-y-1.5`.


https://github.com/RetroAchievements/RAWeb/assets/3984985/a40f4baa-6113-4f24-9020-22d0f3567e63

Jamiras commented on public/userInfo.php:
```diff
@@ -502,113 +502,34 @@ function resize() {
+        ', [
+            'recentlyPlayedCount' => $userMassData['RecentlyPlayedCount'],
+            'recentlyPlayedEntities' => $userMassData['RecentlyPlayed'],
+            'recentAchievementEntities' => $userMassData['RecentAchievements'],
```
I get an exception on this line accessing the profile page for the most recently registered user.

wescopeland replied:
Good catch, fixed in https://github.com/RetroAchievements/RAWeb/pull/1989/commits/243254d177ab6b39570841da23932f30ebec9ab6 by setting some sane fallbacks.

wescopeland commented:
Safe to merge once #1999 is merged, as it is the only outstanding gating factor on shipping the beaten games feature.
